hikari/commands.py

```diff
@@ -172,6 +190,9 @@ class Command(snowflakes.Unique):
     id: snowflakes.Snowflake = attr.field(hash=True, repr=True)
     # <<inherited docstring from Unique>>.
 
+    type: CommandType = attr.field(default=CommandType.CHAT_INPUT, hash=True, repr=True)
```

Comment:
As with other type tagged enums, probably should consider splitting this into like 2 or 3 separate classes to avoid stuff like redundant fields and unnecessarily vague type hints. Also consider having two separate option types here as well